xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-api/src/main/java/org/xwiki/model/internal/document/DefaultDocumentAuthors.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.model.internal.document;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.xwiki.model.document.DocumentAuthors;
+import org.xwiki.user.UserReference;
+
+/**
+ * Default implementation of {@link DocumentAuthors}.
+ *
+ * @version $Id$
+ * @since 13.10RC1
+ */
+public class DefaultDocumentAuthors implements DocumentAuthors
+{
+    private UserReference contentAuthor;
+    private UserReference metadataAuthor;
+    private UserReference displayedAuthor;
+
+    /**
+     * Default empty constructor.
+     */
+    public DefaultDocumentAuthors()
+    {
+    }
+
+    /**
+     * Default constructor cloning the given document authors.
+     *
+     * @param documentAuthors the authors information to be cloned.
+     */
+    public DefaultDocumentAuthors(DocumentAuthors documentAuthors)
+    {
+        this.contentAuthor = documentAuthors.getContentAuthor();
+        this.metadataAuthor = documentAuthors.getMetadataAuthor();
+        this.displayedAuthor = documentAuthors.getDisplayedAuthor();
+    }
+
+    /**
+     * Specify the author of the content of the document: this author is only responsible to the content, and not to
+     * other information of the document such as xobjects.
+     *
+     * @param contentAuthor the author of the content of the document.
+     * @return the current instance for Builder pattern.
+     */
+    public DefaultDocumentAuthors setContentAuthor(UserReference contentAuthor)
```

Comment:
I'm wondering if `contentAuthor(UserReference contentAuthor)` would not make more sense for what seems to be a fluent API. In my experience, the return of a "set*" when it's not void is generally more the previous value. 

Now, I'm not sure we really need a fluent API here.

Reply:
> I'm wondering if contentAuthor(UserReference contentAuthor) would not make more sense for what seems to be a fluent API. In my experience, the return of a "set*" when it's not void is generally more the previous value.

I'm personally not a big fan of APIs not starting with `set` for setters, even in fluent APIs. Just because when using auto-completion it's easier to spot what you need to set like that. But probably a matter of taste.

> Now, I'm not sure we really need a fluent API here.

The idea here is to always be able to call a `new DefaultDocumentAuthors().setXXX()` when you need to call `XWikiDocument#setAuthors` to simplify writing it.